internals-js/src/operations.ts

```diff
@@ -855,14 +928,14 @@ export class NamedFragments {
       if (!names.includes(fragment.name)) {
         // We want to keep that fragment. But that fragment might use a fragment we
         // remove, and if so, we need to expand that removed fragment.
-        const updatedSelection = fragment.selectionSet.expandFragments(names, false);
-        const newFragment = updatedSelection === fragment.selectionSet
+        const updatedSelectionSet = fragment.selectionSet.expandFragments(names, newFragments);
+        const newFragment = updatedSelectionSet === fragment.selectionSet
           ? fragment
-          : new NamedFragmentDefinition(fragment.schema(), fragment.name, fragment.typeCondition, updatedSelection);
+          : fragment.withUpdatedSelectionSet(updatedSelectionSet);
         newFragments.add(newFragment);
       }
     }
-    return newFragments;
+    return newFragments.isEmpty() ? undefined : newFragments;
```

Comment:
All the rest of code uses `undefined` when there is no named fragments (which is to make it obvious when there is a no fragments are that's common and avoiding allocating empty objects), so returning an empty object here was making things inconsistent and this tripped me while debugging so I make it more consistent overall.

Of course, a different choice to make things consistent would have been to flip all the other places in the code to never use `undefined` and instead create empty objects when we need it. It would have been more overall change here however, which is the short reason why I changed this one place instead. Though I'll point that the query planner is already pretty memory hungry and I've seen empty objects like that genuinely show up in heap dumps before, courtesy of javascript, so I also don't think that using empty objects is unequivocally 
 better.